File: src/javascript/user/addressbook/create-address-book.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

if (process.env.FAXCORE_CONFIRM_DESTRUCTIVE !== 'true') {
  throw new Error('Set FAXCORE_CONFIRM_DESTRUCTIVE=true to create an address book.');
}

const client = new FaxCoreClient();
const result = await client.post('/api/addressbook', {
  addressBookName: process.env.FAXCORE_ADDRESS_BOOK_NAME || 'Sample Address Book'
});

printResult('Create address book', result);

File: src/dotnet/FaxCore.ApiSamples/Samples/User/AddressBook/CreateAddressBook.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.AddressBook
{
    internal static class CreateAddressBook
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            RequireDestructiveConfirmation();
            return client.PostAsync("/api/addressbook", new { addressBookName = config.AddressBookName });
        }

        private static void RequireDestructiveConfirmation()
        {
            if (!string.Equals(Environment.GetEnvironmentVariable("FAXCORE_CONFIRM_DESTRUCTIVE"), "true", StringComparison.OrdinalIgnoreCase))
            {
                throw new InvalidOperationException("Set FAXCORE_CONFIRM_DESTRUCTIVE=true to create an address book.");
            }
        }
    }
}
